commands/client_daemon_test.go

```diff
@@ -196,3 +197,40 @@ func TestVoucherPersistenceAndPayments(t *testing.T) {
 	assert.Contains(result, "0\t480000\t20")
 	assert.Contains(result, "0\t720000\t30")
 }
+
+func TestPieceRejectionInProposeStorageDeal(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	assert := assert.New(t)
+	minerDaemon := th.NewDaemon(t,
+		th.WithMiner(fixtures.TestMiners[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[0]),
+		th.DefaultAddress(fixtures.TestAddresses[0]),
+		th.AutoSealInterval("1"),
+	).Start()
+	defer minerDaemon.Shutdown()
+
+	clientDaemon := th.NewDaemon(t,
+		th.KeyFile(fixtures.KeyFilePaths()[1]),
+		th.DefaultAddress(fixtures.TestAddresses[1]),
+	).Start()
+	defer clientDaemon.ShutdownSuccess()
+
+	minerDaemon.RunSuccess("mining", "start")
+	minerDaemon.UpdatePeerID()
+
+	minerDaemon.ConnectSuccess(clientDaemon)
+
+	addAskCid := minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
+	clientDaemon.WaitForMessageRequireSuccess(addAskCid)
+	var byteSlice []byte
+	for i := 0; i < 3000; i++ {
+		byteSlice = append(byteSlice, byte('a'))
```

Comment:
Nit: Since the data is unimportant, you could replace this loop with `byteSlice := make([]byte, 3000)`